# sheib
# introduction
This web page provides source code and simulated datasets mentioned in the article "SHEIB: a novel stochastic-based approach for detecting high-order epistatic interactions using bioinformation in genome-wide association studies". The article has been submitted to BMC Bioinformatics but has not been published. Please don't download these files, if you are not reviewers or editors.<br>
The real dataset (WTCCC dataset) in the article will not be uploaded. Because the dataset was downloaded from https://www.wtccc.org.uk/. We have no right to upload it.
# source code
SHEIB is implemented by C++ and compliled by Cygwin64. The source code has been uploaded. We have also uploaded two executable files (sheib was compiled on linux 64 and sheib.exe was compiled on win7 64).
# introduction to the uploaded files
## sheib.zip:
All files like *.cpp and *.h are source code of SHEIB.<br>
bd_gwas.tped and bd_gwas.tfam are sample examples of real gwas dataset.<br>
data.txt is a sample example of simulated data.<br>
snps_after_2.txt is the gene-mapping data from NCBI.<br>
gene_pairs_after_4.txt is the gene association data from DIP.<br>
hrpd_after_3.txt is the gene association data from HRPD.<br>
mint_after_4.txt is the gene association data from MINT.<br>
reactome_after_4.txt is the gene association data from Reactome.<br>
sheib is an executable file compiled by gcc on Linux 64.<br>
sheib.exe is an executable file compiled by cygwin64 on win7 64.<br>
sheib.cbp, sheib.depend and sheib.layout are files generated by CodeBlocks automatically. Users can use CodeBlocks to open them.<br>
# run on linux 64
compile: g++ -std=c++11 -lpthread *.cpp -o sheib<br>
./sheib -type 0 -cG 0.05 -cGc 0.05 -o -1 -maxGen 4000000 -pb 0.8 -nShow 4000 -seed 0 -rn -1 -cs 0 -in data.txt -out result.txt<br>
./sheib -type 1 -cG 0.05 -cGc 0.05 -o -1 -maxGen 4000000 -pb 0.8 -nShow 4000 -seed 0 -rn -1 -cs 0 -in bd_gwas -out result.txt<br>
./sheib -type 1 -cG 0.05 -cGc 0.05 -o -1 -maxGen 4000000 -pb 0.8 -nShow 4000 -seed 0 -rn -1 -cs 0 -in bd_gwas -out result.txt -SNP2Genes snps_after_2.txt<br>
./sheib -type 1 -cG 0.05 -cGc 0.05 -o -1 -maxGen 4000000 -pb 0.8 -nShow 4000 -seed 0 -rn -1 -cs 0 -in bd_gwas -out result.txt -SNP2Genes snps_after_2.txt -AssociatedGenes hrpd_after_3.txt<br>
# run on win7 64
The folder uploaded is a project of CodeBlocks. You can open the sheib.cbp using CodeBlocks.<br>
sheib -type 0 -cG 0.05 -cGc 0.05 -o -1 -maxGen 4000000 -pb 0.8 -nShow 4000 -seed 0 -rn -1 -cs 0 -in data.txt -out result.txt<br>
sheib -type 1 -cG 0.05 -cGc 0.05 -o -1 -maxGen 4000000 -pb 0.8 -nShow 4000 -seed 0 -rn -1 -cs 0 -in bd_gwas -out result.txt<br>
sheib -type 1 -cG 0.05 -cGc 0.05 -o -1 -maxGen 4000000 -pb 0.8 -nShow 4000 -seed 0 -rn -1 -cs 0 -in bd_gwas -out result.txt -SNP2Genes snps_after_2.txt<br>
sheib -type 1 -cG 0.05 -cGc 0.05 -o -1 -maxGen 4000000 -pb 0.8 -nShow 4000 -seed 0 -rn -1 -cs 0 -in bd_gwas -out result.txt -SNP2Genes snps_after_2.txt -AssociatedGenes hrpd_after_3.txt<br>
# simulated dataset
The simulated data is too large to upload (3.08GB simulated_data.tar.gz). So we only uploaded the DNME3_100 dataset. If you need all the three simulated datasets, you can send email to 2276632042@qq.com (the first author of the article).<br>
